doc/nrf/app_dev/device_guides/hpf/assembly_management.rst

```diff
@@ -0,0 +1,105 @@
+These components have strict timing requirements and are stored in a pre-compiled assembly format to ensure they remain unchanged.
+
+The assembly management CMake functions are integrated into the build process and are executed with every build.
+These functions check for any modifications that could lead to the generation of different assembly files, such as changes in HRT source files or the compiler used.
+Detecting these changes is essential because they can impact the timing accuracy of software peripherals.
+If any changes are detected that could affect the assembly files, you must verify that these differences are within acceptable limits.
+This ensures that the new code maintains the necessary timing accuracy.
+
+To use this feature, you must include the command ``hpf_assembly_install(app "${CMAKE_SOURCE_DIR}/file.c")`` in your application's :file:`CMakeLists.txt` file, where ``file.c`` value is the C file containing only Hard Real-Time (HRT) functions.
+
+Although the assembly management functions are called every time in build time, some of them can also be run independently by using CMake targets in VPR core's application build directory.
+
+See following flow of the build-time assembly management process:
+
+.. IMG TBA
+
+.. _hpf_assembly_management_cmake:
+
+Implementation
+**************
+
+The assembly management performed at build time uses following :ref:`CMake<cmake_options>` functions:
+
+* :c:func:`hpf_assembly_generate`
+* :c:func:`hpf_assembly_check`
+* :c:func:`hpf_assembly_prepare_install`
+* :c:func:`hpf_assembly_target_sources`
+
+hpf_assembly_generate
+=====================
+
+The :c:func:`hpf_assembly_generate` function creates a build target (asm_gen) that generates a temporary assembly file from HRT source files.
+This files is named :file:`file-soc-temp.s` and is located in the Fast Lightweight Peripheral Processor (FLPR) application's build directory when running ``ninja build``.
+In the filename, *soc* represents the System on Chip (SoC) name used in the current build, such as ``nrf54l15``.
+
+hpf_assembly_check
+==================
+
+The :c:func:`hpf_assembly_check` function sets up a build target (asm_check) that is triggered when executing ``ninja build``, following the :c:func:`sdk_assembly_generate` function.
+The target verifies if the :file:`file-soc-temp.s` file in the build directory matches the :file:`file-soc.s` file in the source directory.
+If discrepancies are found, it triggers a CMake error (in user mode) or warning (in developer mode).
+
+hpf_assembly_prepare_install
+============================
+
+The :c:func:`hpf_assembly_prepare_install` function creates a build target (asm_install) that replaces the :file:`file-soc.s` file in the source directory with :file:`file-soc-temp.s` from the build directory.
+
+hpf_assembly_target_sources
+===========================
+
+The :c:func:`hpf_assembly_target_sources` function includes the :file:`file-soc.s` from source directory in the target sources.
```

Comment:
@annwoj this sentence needs to be changed to something like:
```suggestion
The :c:func:`hpf_assembly_target_sources` function includes the :file:`file-soc.s` from source directory (user mode) or the :file:`file-soc-temp.s` from build directory in the target sources.
```
Just to make sure that there is an indication of which assembly files are included in build depending on user/developer build mode. I forgot to add it myself.